src/GraphQL/Types/Scalars/IdGraphType.cs

```diff
@@ -24,7 +24,7 @@ public IdGraphType()
         }
 
         /// <inheritdoc/>
-        public override object Serialize(object value) => value?.ToString();
+        public override object Serialize(object value) => value.ToString();
```

Comment:
Why remove `?` ?  NRE if `value` is `null`. The same for `ParseValue` and `ToAST`

Reply:
`value` cannot be null.

Reply:
Why? Who guarantees this? This is a public method.

Reply:
We can define this method it any way we want. In this case the definition is that these methods should not be called with a null argument. We can define the result when it is called with an invalid argument. Perhaps “the result is undefined”. Or “it should return null”. Or “it should return null or throw an exception”. Or even “it should throw an argument null exception”.  In this case I think “it should return null or throw an exception” is perfectly acceptable, as that is the definition for any other value that cannot be coerced. Again, ANY invalid value is ALREADY defined as “it should return null or ANY exception”. So why not “null” also? It is just another invalid value. 

That is the definition. People can correctly implement it when they write their implementation, or they can write buggy code. Users of the method should write their code with the assumption that it was written with semantics that match the definition written above. The library is already correctly written to assume the above semantics. 

We can choose to return null or throw an exception in our implementation. We can choose what exception is thrown. ArgumentNullException is most proper given the method semantics. But returning null or throwing NullReferenceException does not break the rules we have previously defined. Unless we change the rules, of course. But there is no reason to change them.

Reply:
In a separate PR perhaps we will change the definition to read that null is a valid value to be supplied, and the method should return a serialized value that represents null. That is a change of the definition, requiring a change of the implementation.

Reply:
OK. We will assume that if NRE arises, then we will deal with this situation separately.

Reply:
Added xml comments to define semantics of methods